src/Functions/URL/netloc.h

```diff
@@ -2,41 +2,128 @@
 
 #include "FunctionsURL.h"
 #include <common/find_symbols.h>
+#include "protocol.h"
+#include <cstring>
+#include <Common/StringUtils/StringUtils.h>
 
 
 namespace DB
 {
 
-struct ExtractNetloc
-{
-    /// We use the same as domain function
-    static size_t getReserveLengthForElement() { return 15; }
-
-    static void execute(Pos data, size_t size, Pos & res_data, size_t & res_size)
+    struct ExtractNetloc
     {
-        res_data = data;
-        res_size = size;
-
-        Pos pos = data;
-        Pos end = pos + size;
-
-        pos = find_first_symbols<'/'>(pos, end);
-        if (end == pos)
-            return;
-
-        /// Strings are zero-terminated.
-        bool has_subsequent_slash = pos[1] == '/';
-        if (!has_subsequent_slash)
-            return;
-        /// Search for next slash or question mark
-        /// Note than currently the netloc function doesn't support
-        /// if we have a question mark as an username or password.
-        /// This choice has been made for not hurting performance with still taking into
-        /// acount URL without a slash in the end of a query, with a query string.
-        res_data = pos + 2;
-        res_size = find_first_symbols<'/', '?'>(pos + 2, end) - res_data;
-    }
-};
+        /// We use the same as domain function
+        static size_t getReserveLengthForElement() { return 15; }
+
+        static inline StringRef getNetworkLocation(const char * data, size_t size)
+        {
```

Comment:
Looks like copy-pasted getURLHost (if I'm not missing anything).

Maybe `getURLHost` can accept `userinfo` parameter and return hostname with userinfo or not instead based on this (and then all `getNetworkLocation` can be replaced with `getURLHost(userinfo=true)`?

Reply:
We have some difference between the `getURLHost`, even if indeed the function was used as a base.

On `getURLHost`, we stop when we have a `?`, `/`, `@` or `:`, but this kind of information can be totally considered as valid, when we have a password for example.

We could possibly use it, but we should have on mind than the `domain` function will be significantly slower, if we doesn't have any identification on the url, we would need to parse entirely the URL as we cannot detect if we are currently parsing an user/password or the domain

Reply:
>On getURLHost, we stop when we have a ?, /, @ or :, but this kind of information can be totally considered as valid, when we have a password for example.

Forgot about this, ok